arm-network/2017-03-01/swagger/azureWAN.json

```diff
@@ -0,0 +1,3345 @@
+                        "description": "Request successful. Returns the details of the ExpressRoute connection retrieved.",
+                        "schema": {
+                            "$ref": "#/definitions/ExpressRouteConnection"
+                        }
+                    },
+                    "404": {
+                        "description": "ExpressRoute Connection doesn't exist"
+                    }
+                }
+            },
+            "put": {
+                "operationId": "AzureWANExpressRouteConnection_CreateOrUpdate",
+                "description": "Creates an ExpressRoute connection to a scalable ExpressRoute gateway if it doesn't exist else updates the existing connection.",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/HubParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/GatewayParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ConnectionNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "name": "parameters",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/ExpressRouteConnectionPutOrPatchParameters"
+                        },
+                        "description": "Parameters supplied to create or update a ExpressRoute connection to a scalable ExpressRoute gateway."
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Request successful. Returns the details of the ExpressRoute connection created or updated.",
+                        "schema": {
+                            "$ref": "#/definitions/ExpressRouteConnection"
+                        }
+                    },
+                    "202": {
+                        "description": "Request received successfully. ExpressRoute connection is not yet provisioned; follow the Location header to poll for final outcome."
+                    },
+                    "400": {
+                        "description": "Bad Request. Failed to validate."
+                    }
+                }
+            },
+            "patch": {
+                "operationId": "AzureWANExpressRouteConnection_Update",
+                "description": "Updates an ExpressRoute connection.",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionI
```

Comment:
`scalableVpnGateway` seems like a valid resource then why is this service not using the model in the array here? 

Customers have complained a lot about the pain of traversing things by names, instead it'd greatly simplify experience bf at least we can give then `id` of the subResource then it's easy to get that resource when required via `Microsoft.Resource` service.